applications/IgaApplication/custom_elements/shell_3p_element.cpp

```diff
@@ -79,15 +79,29 @@ namespace Kratos
         if (mConstitutiveLawVector.size() != r_number_of_integration_points)
             mConstitutiveLawVector.resize(r_number_of_integration_points);
 
-
-        for (IndexType point_number = 0; point_number < mConstitutiveLawVector.size(); ++point_number) {
+        for (IndexType point_number = 0; point_number < r_number_of_integration_points; ++point_number) {
             mConstitutiveLawVector[point_number] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
             mConstitutiveLawVector[point_number]->InitializeMaterial(r_properties, r_geometry, row(r_N, point_number));
         }
 
         KRATOS_CATCH("");
     }
 
+    void Shell3pElement::FinalizeSolutionStep(const ProcessInfo& rCurrentProcessInfo)
+    {
+        // Reading integration points
+        const GeometryType& r_geometry = GetGeometry();
+        const Properties& r_properties = GetProperties();
+        const auto& N_values = r_geometry.ShapeFunctionsValues();
+
+        // Reading integration points
+        const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints();
+
+        for (IndexType point_number = 0; point_number < mConstitutiveLawVector.size(); ++point_number) {
+            mConstitutiveLawVector[point_number]->FinalizeSolutionStep(r_properties, r_geometry, row(N_values, point_number), rCurrentProcessInfo);
```

Comment:
This is long deprecated and removed in #7691 

please use the new method

Reply:
indeedily. Thanks!!

Whats about the initialize material, do we keep that beauty?

Reply:
Well you should not use the deprecated methods :)

Reply:
yeah but that one isn't deprecated [yet]

Reply:
I checked and I think there is no equivalent with `ConstitutiveLaw` parameters yet.
Hence I would leave it :)